### PR TITLE
Migrate io.zeebe.broker.engine tests

### LINKED ISSUE #2552: Migrate io.zeebe.broker.engine.* tests
Migrate all tests which are part of the package `io.zeebe.broker.engine` to the engine module and move them in a corresponding package, related to the processor.

blocks #2461

### PR DESCRIPTION
**Pull request by ChrisKujawa**
closes #2552

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c454123f606a1721c19df6c4ee2a317b2dfe01f4
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

File: engine/src/test/java/io/zeebe/engine/processor/workflow/WorkflowAssert.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public static void assertWorkflowInstanceRecord(
       String elementId,
       Record<WorkflowInstanceRecordValue> workflowInstanceCanceledEvent) {
     Assertions.assertThat(workflowInstanceCanceledEvent.getValue())
-        .hasBpmnProcessId(PROCESS_ID)
+        .hasBpmnProcessId(elementId)
```
Comment (MiguelPires) [inline engine/src/test/java/io/zeebe/engine/processor/workflow/WorkflowAssert.java]: Why this change? I realize this method is only used by the method above but this seems strange

Comment (Zelldon) [reply engine/src/test/java/io/zeebe/engine/processor/workflow/WorkflowAssert.java]: This was a left over, sorry. See my latest commit I removed this class

**Review by miguelpires — APPROVED** [review]

**Comment by Zelldon** [comment]
Bors r+

**Comment by Zelldon** [comment]
Bors r+

**Comment by Zelldon** [comment]
Bors r+
